vyper/builtins/functions.py

```diff
@@ -824,27 +814,22 @@ class ECMul(BuiltinFunction):
 
     @process_inputs
     def build_IR(self, expr, args, kwargs, context):
-        placeholder_node = IRnode.from_list(
-            context.new_internal_variable(BytesT(128)), typ=BytesT(128), location=MEMORY
-        )
+        buf_t = get_type_for_exact_size(96)
 
-        with args[0].cache_when_complex("a") as (b1, a), args[1].cache_when_complex("b") as (b2, b):
-            o = IRnode.from_list(
-                [
-                    "seq",
-                    ["mstore", placeholder_node, _getelem(a, 0)],
-                    ["mstore", ["add", placeholder_node, 32], _getelem(a, 1)],
-                    ["mstore", ["add", placeholder_node, 64], b],
-                    [
-                        "assert",
-                        ["staticcall", ["gas"], 7, placeholder_node, 96, placeholder_node, 64],
-                    ],
-                    placeholder_node,
-                ],
-                typ=SArrayT(UINT256_T, 2),
-                location=MEMORY,
-            )
-            return b2.resolve(b1.resolve(o))
+        buf = context.new_internal_variable(buf_t)
+
+        ret = ["seq"]
+
+        dst0 = IRnode.from_list(buf, typ=SArrayT(UINT256_T, 2), location=MEMORY)
+        ret.append(make_setter(dst0, args[0]))
+
+        dst1 = IRnode.from_list(buf + 64, typ=UINT256_T, location=MEMORY)
+        ret.append(make_setter(dst1, args[1]))
+
+        ret.append(["assert", ["staticcall", ["gas"], 7, buf, 128, buf, 64]])
```

Comment:
Shouldn't the precompile only take 96 bytes of input ?
```suggestion
        ret.append(["assert", ["staticcall", ["gas"], 7, buf, 96, buf, 64]])
```

Reply:
Ah yea, seems I changed the buffer size but not the call input